lib/rouge/lexers/scala.rb

```diff
@@ -47,12 +47,38 @@ class Scala < RegexLexer
         rule %r(/\*), Comment::Multiline, :comment
 
         rule /@#{idrest}/, Name::Decorator
+
+        rule /(def)(\s+)(#{idrest}|#{op}+|`[^`]+`)(\s*)/ do
+          groups Keyword, Text, Name::Function, Text
+        end
+
+        rule /(val)(\s+)(#{idrest}|#{op}+|`[^`]+`)(\s*)/ do
+          groups Keyword, Text, Name::Variable, Text
+        end
+
+        rule /(this)(\n*)(\.)(#{idrest})/ do
+          groups Keyword, Text, Operator, Name::Property
+        end
+
+        rule /(#{idrest}|_)(\n*)(\.)(#{idrest})/ do
+          groups Name::Variable, Text, Operator, Name::Property
+        end
+
+        rule /#{upper}#{idrest}\b/, Name::Class
+
+        rule /(#{idrest})(#{whitespace}*)(\()/ do
+          groups Name::Function, Text, Operator
+        end
+
+        rule /(\.)(#{idrest})/ do
+          groups Operator, Name::Property
+        end
+
         rule %r(
           (#{keywords.join("|")})\b|
           (<[%:-]|=>|>:|[#=@_\u21D2\u2190])(\b|(?=\s)|$)
         )x, Keyword
-        rule /:(?!#{op})/, Keyword, :type
-        rule /#{upper}#{idrest}\b/, Name::Class
```

Comment:
Is there any reason to remove this rule? I understand that it's not required for class names to begin with an uppercase letter but, given it's [the naming convention](https://docs.scala-lang.org/style/naming-conventions.html#classestraits), this seems like a good heuristic that will produce more expressive highlighting in most cases.

Reply:
It's been a while since I've written this! I'll take a look later today and see if I can recollect my thought process here. Thanks for pushing this project forward! 🎉

Reply:
On second look, I actually just rearranged it 😛 . It's here now: https://github.com/rouge-ruby/rouge/pull/1040/files#diff-26a3df96d71bb9ce8c56b77fa7c97f21R67.